globalsat_sport.cc

```diff
@@ -0,0 +1,742 @@
+{
+	CommandWhoAmI = 0xBF,
+	CommandGetSystemInformation = 0x85,
+	CommandGetSystemConfiguration = 0x86,
+
+	CommandSetSystemConfiguration = 0x96,
+	CommandSetSystemInformation = 0x98,
+	CommandGetScreenshot = 0x83,
+
+	CommandGetWaypoints = 0x77,
+	CommandSendWaypoint = 0x76,
+	CommandDeleteWaypoints = 0x75,
+
+	CommandSendRoute = 0x93,
+	CommandDeleteAllRoutes = 0x97,
+
+	CommandGetTrackFileHeaders = 0x78,
+	CommandGetTrackFileSections = 0x80,
+	CommandGetNextTrackSection = 0x81,
+	//CommandReGetLastSection = 0x82,
+	CommandId_FINISH = 0x8A,
+	CommandSendTrackStart = 0x90,
+	CommandSendTrackSection = 0x91,
+
+	HeaderTypeLaps = 0xAA,
+	HeaderTypeTrackPoints = 0x55,
+
+	ResponseInsuficientMemory = 0x95,
+	ResponseResendTrackSection = 0x92,
+	ResponseSendTrackFinish = 0x9A
+} globalsat_commands_e;
+
+
+//Pack structures since they should conform to serial data spec
+#pragma pack(1)
+
+typedef struct tagDATE
+{
+	uint8_t Year;
+	uint8_t Month;
+	uint8_t Day;
+} gh_date;
+
+typedef struct tagTIME
+{
+	uint8_t Hour;
+	uint8_t Minute;
+	uint8_t Second;
+} gh_time;
+
+typedef struct tagTRAINHEADER
+{
+	gh_date dateStart;
+	gh_time timeStart;
+	uint32_t TotalPoint;		//6-9
+	uint32_t TotalTime;		//10-13
+	uint32_t TotalDistance;	//14-17
+	uint16_t LapCnts;		//18-19
+	union
+	{
+		uint32_t Index;
+		uint32_t StartPt;
+	} gh_ptrec;			//20-23
+	union
+	{
+		uint32_t LapIndex;
+		uint32_t EndPt;
+	} gh_laprec;			//24-27
+	uint8_t DataType;		//28
+} gh_trainheader;
+
+typedef struct tagDB_TRAIN
+{
+	gh_date dateStart;
+	gh_time timeStart;
+	uint32_t TotalPoint;		//6-9
+	uint32_t TotalTime;		//10-13
+	uint32_t TotalDistance;	//14-17
+	uint16_t LapCnts;		//18-19
+	union
+	{
+		uint32_t Index;
+		uint32_t StartPt;
+	} gh_ptrec;			//20-23
+	union
+	{
+		uint32_t LapIndex;
+		uint32_t EndPt;
+	} gh_laprec;			//24-27
+	uint8_t MultiSport;		//28 on/off
+	uint16_t Calory;		//29-30
+	uint32_t MaxSpeed;
+	uint8_t MaxHeart;
+	uint8_t AvgHeart;
+
+	uint16_t Ascent;
+	uint16_t Descent;
+
+	int16_t MinAlti;
+	int16_t MaxAlti;
+	uint16_t AvgCadns;
+	uint16_t BestCadns;
+	uint16_t AvgPower;
+	uint16_t MaxPower;
+	uint8_t Sport1;
+	uint8_t Sport2;
+	uint8_t Sport3;
+	uint8_t Sport4;
+	uint8_t Sport5;
+} gh_db_train;
+
+typedef struct tagDB_LAP
+{
+	uint32_t AccruedTime;
+	uint32_t TotalTime;
+	uint32_t TotalDistance;
+	uint16_t Calory;
+	uint32_t MaxSpeed;
+	uint8_t MaxHeart;
+	uint8_t AvgHeart;
+	int16_t MinAlti;
+	int16_t MaxAlti;
+	uint16_t AvgCadns;
+	uint16_t BestCadns;
+	uint16_t AvgPower;
+	uint16_t MaxPower;
+	uint8_t MultiSportIndex;
+	uint32_t StartPt;
+	uint32_t EndPt;
+} gh_db_lap;
+
+typedef struct tagRECPOINT
+{
+	uint32_t Latitude;
+	uint32_t Longitude;
+	int16_t Altitude;
+	uint32_t Speed;
+	uint8_t HeartRate;
+	uint32_t IntervalTime;
+	uint16_t Cadence;
+	uint16_t PwrCadence;
+	uint16_t Power;
+} gh_recpoint;
+
+//#pragma options align=reset
+
+static void
+serial_init (const char *fname)
+{
+	if (serial_handle = gbser_init (fname), NULL == serial_handle) {
```

Comment:
I'll fix it with astyle, but we generally write init() instead of init ().  Yes, I know that whitespace code convention arguments are a hassle.   I'll fix before submit if necessary.